source/_posts/2015-10-17-headless-firefox-in-ubuntu-on-virtual-box-for-cucumber-testing.markdown

```diff
@@ -0,0 +1,78 @@
+---
+layout: post
+title: "Headless firefox in ubuntu on virtual box for cucumber testing"
+date: 2015-10-17 14:12:56 +0200
+comments: true
+author: Ivica Lakatoš
+categories: 
+  - Ruby and Rails
+  - Testing
+tags: 
+  - ruby
+  - rails
+  - vagrant
+  - virtual box
+  - ubuntu
+  - cucumber
+  - selenium-webdriver
+published: true
+---
+
+If you use [Vagrant](http://www.vagrantup.com/downloads.html) and [VirtualBox](https://www.virtualbox.org/) (e.g. some Ubuntu package) to build your Rails app and you want to test it with cucumber scenarios, this is right post for you. 
+
+Everything goes well until you do not need `@javascript` flag for your cucumber scenario. `@javascript` uses a javascript-aware system to process web requests (e.g., Selenium) instead of the default (non-javascript-aware) webrat browser.
+
+### You will need mozilla firefox
+
+Selenium WebDriver is flexible and let you run selenium headless in servers with no display. But in order to run, Selenium needs to launch a browser. If there are no display to the machine, the browsers are not launched. So in order to use selenium, you need to fake a display and let selenium and the browser thinks they are running in a machine with a display.
```

Comment:
@Ikera related to first paragraph where I suggested to add a note that Ubuntu is running without X server and GUI, this paragraph can be rephrased a bit. Something along the lines... Since Ubuntu is running without a X server Selenium cannot start Firefox because it requires an X server...